### DISCUSSION TITLE
文件和文件夹md5工具-md5any

### DISCUSSION THREAD

**Discussion by caibingcheng** [caibingcheng/comment#302] [Announcements]
https://imcbc.cn/202210/md5fileandic-md5any/

**Comment by ryan4yin** [comment]
哈哈，思路很好，不过其实不需要这么麻烦，github action 本身就有 path 参数，可以看看我这个例子：

https://github.com/ryan4yin/waline-comments-backup/blob/main/.github/workflows/backup.yaml

文件夹的语法也是支持的，详见官方文档：

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

**Comment by caibingcheng** [comment]
> @ryan4yin
> 哈哈，思路很好，不过其实不需要这么麻烦，github action 本身就有 path 参数，可以看看我这个例子：
> 
> https://github.com/ryan4yin/waline-comments-backup/blob/main/.github/workflows/backup.yaml
> 
> 文件夹的语法也是支持的，详见官方文档：
> 
> https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

学到了！但是场景有点不一样。上面这个说是path在push和pr下用的，我的策略是只有schedule触发，push和pr不触发，所以需要在schedule的时候检测目标路径/文件是否有改动。

用schedule触发是参考了之前杜老师的策略

**Comment by ryan4yin** [comment]
@caibingcheng 喔喔是这样

**Comment by penndu** [comment]
手动命令还是麻烦，如果有GUI工具可以直接对比就方便了！
